### PR TITLE
Add mmsC3Plot directive

### PR DESCRIPTION
**Pull request by miyakowilson**
The mmsC3Plot directive works when C3Plot is defined using the mdk's View Method Diagram with C3Plot templates with TableStructure (cells of slots).

### CODE REVIEW

File: src/directives/mmsC3Plot.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  var mmsChartLink = function(scope, element, attrs, mmsViewCtrl) {
+
+    var c3 = $window.c3;
+    scope.rowHeaders=[]; //not null when render is called 1st time.      
+    var d3 = $window.d3;  
+    var svg = d3.select(element[0])
+      .append('div');
+    
+    var processed = false;
+    var ws = scope.mmsWs;
```
Comment (dlamoris) [inline src/directives/mmsC3Plot.js]: there's no more ws and version

File: src/directives/mmsC3Plot.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    svg.selectAll('*').remove();
+	
+  	var has_column_header;
+    var start_index; //0 if column header is included as data, -1 if column header is not included as data
+    var c3_data=[];
+
+      var c3options;    
+      if ( scope.plot.options === undefined || scope.plot.options.length === 0 ){
+        if (scope.tableColumnHeadersLabel && scope.tableColumnHeadersLabel.length !== 0){
+          c3options = JSON.parse('{"data": {"x": "x", "type": "line"},"axis" : {"x": {"type":"category", "tick":{"centered":true}}}}');
```
Comment (dlamoris) [inline src/directives/mmsC3Plot.js]: don't need to use JSON.parse for this, just set it as the object

File: src/directives/mmsC3Plot.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  };//end of render
+
+    scope.$watch('datavalues', function(newVals, oldVals) {
+        return scope.render();
+    }, true);
+
+   
+    var reqOb = {tableData: scope.plot.table, projectId: projectId, refId: refId, commitId: commitId};
+
+    TableService.readTable (reqOb, ws, version)
```
Comment (dlamoris) [inline src/directives/mmsC3Plot.js]: can remove ws and version

File: src/services/TableService.js [hunk 4]
```diff
@@ -16,9 +16,10 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
      *
      * @param {string} mmsEid SysML ID of a table instance
      */
-    var readTableCols = function(mmsEid, ws, version) {
+    var readTableCols = function(reqOb, ws, version) {
```
Comment (dlamoris) [inline src/services/TableService.js]: don't need ws and version

File: src/services/TableService.js [hunk 5]
```diff
@@ -16,9 +16,10 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
       var deferred = $q.defer();
-      ElementService.getElement(mmsEid, false, ws, version)
+      //ElementService.getElement(mmsEid, false, ws, version)
+      ElementService.getElement(reqOb, 1, true)
```
Comment (dlamoris) [inline src/services/TableService.js]: last argument can be false or just leave off, if false it'll get from the cache if availalbe

File: src/services/TableService.js (hunk 4 above)
Comment (dlamoris) [inline src/services/TableService.js]: is this function still needed? the code inside is still using crushinator (2.x) json schema?

File: src/services/TableService.js [hunk 6]
```diff
@@ -98,29 +99,54 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
+        console.log(data);
+
         var tableTitles = []; //used only for display
         var tableIds = []; //used as filter id
         var tableColumnHeadersLabels=[];
         var numOfDataColumn = []; //to know the number of data columns in case not column header lables.
         var numOfRowHeadersPerTable = [];
         var rowHeadersMmsEid = [];
-        var dataValuesMmmEid =[];
+        rowHeadersMmsEid.elementIds =[];
```
Comment (dlamoris) [inline src/services/TableService.js]: rowHeaderMmsEid should be {} instead of []

File: src/services/TableService.js [hunk 7]
```diff
@@ -98,29 +99,54 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
+        rowHeadersMmsEid.projectId = reqOb.projectId;
+        rowHeadersMmsEid.refId = reqOb.refId;
+        rowHeadersMmsEid.commitId = tableIds;
```
Comment (dlamoris) [inline src/services/TableService.js]: should be reqOb.commitId?

File: src/services/TableService.js [hunk 8]
```diff
@@ -98,29 +99,54 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
+        var dataValuesMmmEid=[];
```
Comment (dlamoris) [inline src/services/TableService.js]: should be {}

File: src/services/TableService.js [hunk 9]
```diff
@@ -98,29 +99,54 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
+        dataValuesMmmEid.commitId = reqOb.commitId;
+
         var columnCounter;
         var i, j, k;
+
+     
         if ( data._contents !==  undefined){ //use contents if exist
         //if ( data.specialization.contains ===  undefined){  
           var tempMmsEid = [];
+          tempMmsEid.elementIds = [];
```
Comment (dlamoris) [inline src/services/TableService.js]: tempMmsEid should be {}

File: src/services/TableService.js [hunk 10]
```diff
@@ -258,11 +248,86 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
...
+        } //end of readTableCommon function
+     
+      return deferred.promise;
+    }; //end of readTable
 
     return {
         readTables: readTables,
         toValidId: toValidId,
-        readTableCols: readTableCols
+        readTableCols: readTableCols,
```
Comment (dlamoris) [inline src/services/TableService.js]: is readTables and readTableCols used anywhere? seems like only readTable is used now, can remove the other two

Comment (miyakowilson) [inline src/services/TableService.js]: They are used in d3 charts.  The d3 charts are still included in src\directives\charts.  GroupHorizontalBarChart and LineGraph can be replaced with c3.  I will remove the code for now and update with d3 charts later.

**Pull request merged**